src/Microsoft.ML.AutoML/TrainerExtensions/MultiTrainerExtensions.cs

```diff
@@ -229,4 +229,22 @@ public PipelineNode CreatePipelineNode(IEnumerable<SweepableParam> sweepParams,
                 columnInfo.LabelColumnName, columnInfo.ExampleWeightColumnName);
         }
     }
+
+    internal class ImageClassificationExtension : ITrainerExtension
+    {
+        public IEnumerable<SweepableParam> GetHyperparamSweepRanges() => null;
+
+        public ITrainerEstimator CreateInstance(MLContext mlContext, IEnumerable<SweepableParam> sweepParams,
+            ColumnInformation columnInfo)
+        {
+            var options = TrainerExtensionUtil.CreateOptions<Dnn.ImageClassificationTrainer.Options>(null, columnInfo.LabelColumnName);
```

Comment:
Would it be possible for the ImageClassifier to have a GPU setting? 

`UseGPU = { Auto, On, Off }`

Options.
* `Auto` will try the GPU and fail over to the CPU (default)
* `On` will use the GPU the throw if not possible. 
* `Off` will use only the CPU.

Reply:
Not in the current release. You will need to add GPU nuget at compile time to use it. There are ways around to auto detection if the user has GPU, CUDA and CUDANN installed and then copying binaries around but it will be after November release. Please also note tensorflow also works the same way as us currently.

---
In reply to: [339330427](https://github.com/dotnet/machinelearning/pull/4395#discussion_r339330427) [](ancestors = 339330427)

Reply:
Ok. Thanks. 

Being AutoML, we should handle this for the user. We can handle either at the `ImageClassificationTrainer` level, or at the AutoML level. Either will work, though the former has benefits for users directly using the `ImageClassificationTrainer`.